### PR TITLE
New feature allowing special value @currentProcess for InputTag proce…

### PR DESCRIPTION
**Pull request by wddgit**
…ss name

### CODE REVIEW

**Comment by wddgit** [comment]
please test

**Comment by wddgit** [comment]
One negative comment about this new feature. If it is used and then one wants to split a single process into two processes, then a module configured this way might no longer be able to find its input. If the module was configured with an empty process name, then it would continue to succeed in this situation. I think this is why we decided not to implement this a long time ago. But I can imagine there will be some situations where this feature is useful.

**Comment by wddgit** [comment]
@fwyzard This would replace PR #19285. It implements the same feature in a completely different way.

File: FWCore/Framework/src/EDConsumerBase.cc [hunk 1]
```diff
@@ -76,6 +76,8 @@ EDConsumerBase::consumesCollector() {
 
 static const edm::InputTag kWasEmpty("@EmptyLabel@");
 
+static const std::string kCurrentProcess("@currentProcess");
```
Comment (Dr15Jones) [inline FWCore/Framework/src/EDConsumerBase.cc]: How about moving this to be a class static of `InputTag` just like `kSkipCurrentProcess`?

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: FWCore/Framework/src/EDConsumerBase.cc (hunk 1 above)
Comment (wddgit) [inline FWCore/Framework/src/EDConsumerBase.cc]: That would be better. I will change this.

**Comment by wddgit** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by wddgit** [comment]
The unit test that failed is not related to this PR. It is already failing in the IBs.

**Comment by davidlange6** [comment]
merge

**Comment by fwyzard** [comment]
@wddgit thank you for this.

Can I ask for a 9.2.x backport, in order to start using it at HLT ?

**Comment by wddgit** [comment]
@fwyzard I will backport this to 9.2.X soon. There are couple of other things I need to finish up first.

**Comment by fwyzard** [comment]
Thank you,
.A

**Pull request merged**